### DISCUSSION TITLE
Tsp, find a priority delivery strategy, customers on the same route do not pass through separately

### DISCUSSION THREAD

**Discussion by maxiulin** [google/or-tools#3120] [Routing (and legacy CP) questions]
**What language and solver does this apply to?**
Python,Routing

**Describe the problem you are trying to solve.**
Hi,In the process of tsp optimization, the vehicle passes twice on a road and is sent to the customer both times. Is there a strategy to ensure that although it passes twice, only the first time is sent to the customer?

**Additional context**
![image](https://user-images.githubusercontent.com/27402675/152719153-1bcfb153-ed07-4b89-8566-fb9f8ce04104.png)

**Comment by Mizux** [comment]
not sure the vehicle "pass twice", i.e. internally the solver can only visit a location once.

If you look at the solution did you see two times this location ?
I think it's a visual artifact in your rendering not a "double visit"

**Comment by maxiulin** [comment]
Thanks for the reply, I'm pretty sure it's not a visual artifact , the vehicle "pass twice", it's not that the same customer is visit twice, but all customers on the same road pass through separately, in fact, it only needs to visit all customers once, Returning empty vehicle for twice
![QQ截图20220207160814](https://user-images.githubusercontent.com/27402675/152748998-314a810d-2723-47e2-8f81-efb04fed258f.png)
for example
one road,two lines
I want to pass A->B->C
not A->C->B
Although the distance is the same

**Reply by Mizux** [reply]
then in your distance matrix only allow the transit
`A -> B`, `B -> C`, `C -> B`, `B -> A` and not `A -> C` or `C -> A` ?

note: you can disable arc (X, Y) by returning a transit (distance) cost superior to the vehicle (distance) max capacity.

**Comment by bjmwang** [comment]
He might want to ask the issues about visiting customers on a dead end street.

For all the customers on the dead end street, he want to visit them successively while driving into the street,  and visit no customer while driving out.
